source/common/http/async_client_impl.cc

```diff
@@ -134,6 +134,9 @@ AsyncStreamImpl::AsyncStreamImpl(AsyncClientImpl& parent, AsyncClient::StreamCal
   route_ = std::move(*route_or_error);
   stream_info_.dynamicMetadata().MergeFrom(options.metadata);
   stream_info_.setIsShadow(options.is_shadow);
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.async_client_scheme_header_fix")) {
+    stream_info_.setShouldSchemeMatchUpstream(true);
```

Comment:
LGTM in general.

Can we add a unit test to verify this change, like with this to be true, the scheme sent to the upstream is indeed matching the upstream?

Reply:
Good idea about tests!
/wait